sw/device/tests/sim_dv/data_integrity_escalation_reset_test.c

```diff
@@ -156,83 +277,114 @@ void ottf_external_isr(void) {
+        kFaultTarget);
+
+  // Increment the nmi interrupt count.
+  uint32_t exception_count =
+      flash_ctrl_testutils_counter_get(kCounterException);
+  if (exception_count > kMaxInterrupts) {
+    LOG_INFO("Saturating exception counter at %d", exception_count);
+  } else {
+    flash_ctrl_testutils_counter_set_at_least(
+        &flash_ctrl_state, kCounterException, exception_count + 1);
   }
-  flash_ctrl_testutils_counter_set_at_least(
-      &flash_ctrl_state, kCounterInterrupt, interrupt_count + 1);
 
-  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic, kPlicTarget, &irq_id));
+  rv_core_ibex_fault_checker(true);
 
-  LOG_INFO("Got irq_id 0x%x (%d)", irq_id, irq_id);
+  LOG_INFO("Instr access fault handler exiting");
+}
 
-  top_earlgrey_plic_peripheral_t peripheral = (top_earlgrey_plic_peripheral_t)
-      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
-  if (peripheral == kTopEarlgreyPlicPeripheralAlertHandler) {
-    LOG_INFO("Got expected alert handler interrupt %d", irq_id);
+void ottf_instr_misaligned_fault_handler(void) {
+  LOG_INFO("ottf_instr_misaligned_fault_handler");
+}
 
-    // Disable these interrupts from alert_handler so they don't keep happening
-    // until NMI.
-    uint32_t irq =
-        (irq_id -
-         (dif_rv_plic_irq_id_t)kTopEarlgreyPlicIrqIdAlertHandlerClassa);
-    CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(&alert_handler, irq,
-                                                   kDifToggleDisabled));
+void ottf_illegal_instr_fault_handler(void) {
+  LOG_INFO("ottf_illegal_instr_fault_handler");
+}
 
-    // Disable this interrupt to prevent it from continuously firing. This
-    // should not prevent escalation from continuing.
-    CHECK_DIF_OK(dif_rv_plic_irq_set_enabled(&plic, irq_id, kPlicTarget,
-                                             kDifToggleEnabled));
-  } else {
-    CHECK(false, "Unexpected peripheral %d for ISR", peripheral);
-  }
+void ottf_breakpoint_handler(void) { LOG_INFO("ottf_breakpoint_handler"); }
 
-  // Complete the IRQ by writing the IRQ source to the Ibex specific CC
-  // register.
-  CHECK_DIF_OK(dif_rv_plic_irq_complete(&plic, kPlicTarget, irq_id));
-  LOG_INFO("Load integrity error handler exiting");
+void ottf_load_store_fault_handler(void) {
+  LOG_INFO("ottf_load_store_fault_handler");
 }
 
+void ottf_user_ecall_handler(void) { LOG_INFO("ottf_user_ecall_handler"); }
+
 /**
  * External NMI ISR.
  *
  * Handles NMI interrupts on Ibex for either escalation or watchdog.
  */
 void ottf_external_nmi_handler(void) {
+  dif_rv_core_ibex_nmi_state_t nmi_state = (dif_rv_core_ibex_nmi_state_t){0};
   LOG_INFO("At NMI handler");
 
   // Increment the nmi interrupt count.
-  uint32_t nmi_interrupt_count = flash_ctrl_testutils_counter_get(kCounterNmi);
-  if (nmi_interrupt_count > kMaxInterrupts) {
-    LOG_INFO("Saturating nmi interrupts at %d", nmi_interrupt_count);
+  uint32_t nmi_count = flash_ctrl_testutils_counter_get(kCounterNmi);
```

Comment:
You are right: nmi_interrupt is a "stutter".

Reply:
hehe yes